src/rule.jl

```diff
@@ -292,13 +296,16 @@ macro rule(fform, lambda)
                 q_names,
                 q_types,
                 metatype,
+                addonstype,
                 whereargs
             ) do
                 return quote
                     $(on_index_init)
                     $(m_init_block...)
                     $(q_init_block...)
-                    $(body)
+                    _addons = ()
+                    _message = $(MacroHelpers.remove_returns(body))
```

Comment:
Be careful with removing `return` statements from the whole message function body, as this may break things if a rule contains something like (even without addons)
```
@rule MyNode(...) (args...) = begin
    if condition
        return 1
    end
    return 2
end
```

Reply:
Resolved by [68809b4](https://github.com/biaslab/ReactiveMP.jl/pull/215/commits/68809b4f07b858321bd151dca5b7c3487c7c1ba2). It counts the number of return statements per macro by performing some string matching. Although it is better to be replaced with a proper `walk`, but I didn't manage to get this working.

Reply:
Yeah, that is better, I wonder if its really necessary to use `remove_returns` for the body of the `@rule` macro, but rather only on the body of the `@logscale` macro. I would probably move the `remove_returns` statement from `rule.jl:307` to `rule.jl:417` inside `macro logscale ... end`.

```suggestion
rule:jl:307 _message = body

rule.jl:417 return MacroHelpers.remove_returns(output)
```

I'm afraid that we did not really impose this restriction in the `@rule` macro in the past and it might be breaking. `@logscale` macro, however, is new and it may impose new restrictions on its body (such as no or only 1 return statement).